roles/disko.nix

```diff
@@ -0,0 +1,95 @@
+{ disks ? [ ], ... }:
+let
+  content = {
+    format = "gpt";
+    type = "table";
+    partitions = [
+      {
+        name = "boot";
+        start = "0";
+        end = "1M";
+        part-type = "primary";
+        flags = [ "bios_grub" ];
+      }
+      {
+        name = "mdadm";
+        start = "1M";
+        end = "2G";
+        content = {
+          name = "boot";
+          type = "mdraid";
+        };
+      }
+      {
+        name = "zfs";
+        start = "2G";
+        end = "-16G";
+        content = {
+          pool = "zroot";
+          type = "zfs";
+        };
+      }
+      {
+        name = "swap";
+        start = "-16G";
+        end = "100%";
+        part-type = "primary";
+        content = {
+          type = "swap";
+        };
+      }
+    ];
+  };
+in
+{
+  disk = {
+    disk0 = {
+      device = builtins.elemAt disks 0;
+      type = "disk";
+      inherit content;
+    };
+    disk1 = {
+      device = builtins.elemAt disks 1;
+      type = "disk";
+      inherit content;
+    };
+  };
+  mdadm = {
+    boot = {
+      type = "mdadm";
+      level = 1;
+      metadata = "1.0";
+      content = {
+        type = "filesystem";
+        format = "ext4";
+        mountOptions = [ "nofail" ];
+        mountpoint = "/boot";
+      };
+    };
+  };
+  zpool = {
```

Comment:
`zpool` is a copy of the build04 config with the addition of `mountOptions = [ "nofail" ];` for `root/home`.

Not very familiar with zfs so not sure we still need to set acltype xattr atime or legacy mountpoints like we did in the original disk config?

Reply:
You can keep all the options as they are best practices. You would need to configure either legacy mountpoints or use the zfsutil mount option so that systemd is allowed to mount zfs filesystems: https://github.com/Mic92/dotfiles/blob/b092cce7cc0e3c84dccf84e69f783fceeef09e79/nixos/blob64/hardware-configuration.nix#L14

Reply:
Added the options, `zfsutil` seems to be the default that is set by disko.